src/components/buttons/Button.tsx

```diff
@@ -14,23 +15,58 @@ export const Button = ({
     style,
     labelStyle,
     onClick,
+    status = 'primary',
+    disabled,
     testID,
 }: ButtonProps) => {
     if (mode === 'text') {
         return (
-            <TextButton style={style} labelStyle={labelStyle} onClick={onClick} testID={testID}>
+            <TextButton
+                status={status}
+                style={style}
+                labelStyle={labelStyle}
+                onClick={onClick}
+                testID={testID}
+                disabled={disabled}
+            >
                 {children}
             </TextButton>
         );
     } else if (mode === 'filled') {
         return (
-            <FilledButton style={style} labelStyle={labelStyle} onClick={onClick} testID={testID}>
+            <FilledButton
+                status={status}
+                style={style}
+                labelStyle={labelStyle}
+                onClick={onClick}
+                testID={testID}
+                disabled={disabled}
+            >
                 {children}
             </FilledButton>
         );
+    } else if (mode === 'outlined') {
+        return (
+            <OutlinedButton
+                status={status}
+                style={style}
+                labelStyle={labelStyle}
+                onClick={onClick}
+                testID={testID}
+                disabled={disabled}
+            >
+                {children}
+            </OutlinedButton>
+        );
     } else {
         return (
-            <BaseButton style={style} labelStyle={labelStyle} onPress={onClick} testID={testID}>
+            <BaseButton
```

Comment:
Thinking about it I think it's  not a good idea to fallback to the default react-native-paper's Button component if no `mode` props has been provided.
I cannot tell what it looks like quickly because it is not in the design system.

Would you mind to provide a default value to the `mode` props instead ? Like `filled` ?